src/recommendpolicy/helperFunctions.go

```diff
@@ -92,8 +91,12 @@ func generatePolicy(name, namespace string, labels LabelMap) ([]types.KnoxSystem
 		}
 	}
 
-	return policies, nil
+	if cfg.GetCfgMergePolicy() {
+		GroupedbyLabelPolicies := GroupPoliciesByLabel(policies, ms, name, namespace, labels)
+		return GroupedbyLabelPolicies, nil
```

Comment:
if we are not checking for an error or empty policy, then cant we return the value directly without storing it?
```
if cfg.GetCfgMergePolicy() {
	return GroupPoliciesByLabel(policies, ms, name, namespace, labels), nil
}
```

can we also have a log saying merged `x` hardening policies for deployment `y` in namespace `z` like it says for system policy